static/js/composebox_typeahead.js

```diff
@@ -75,6 +75,8 @@ function query_matches_stream(query, stream) {
 
 // Case-insensitive
 function query_matches_emoji(query, emoji) {
+    // replaces spaces with underscores
+    query = query.split(" ").join("_");
```

Comment:
The `replace` function replaces just the first occurrence of space, whereas we might we dealing with double or triple spaces (such as :japanese post office: ). So I have used split and join.